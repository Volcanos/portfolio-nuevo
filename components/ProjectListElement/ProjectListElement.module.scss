.titleProject {
  color: #6f6f6f;
  font-weight: 800;
  margin-bottom: 0rem;
  margin-top: 1rem;
}
.subtitleProject {
  color: #6f6f6f;
  font-size: 0.8rem;
  margin-bottom: 1rem;
}
.projectListWrapper {
  margin-top: 2rem;
  margin-bottom: 5rem;
  text-align: center;
}
