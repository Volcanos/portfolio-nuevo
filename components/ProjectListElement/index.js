import LinkedImage from '../LinkedImage';
import Anchor from '../Anchor';
import styles from './ProjectListElement.module.scss';

const ProjectListElement = (props) => {
    const {url, title, mainImage, subtitle} = {...props}
    return (
        <div className={styles.projectListWrapper}>
            <LinkedImage url={url} title={title} image={mainImage} prefetch={false} />
            <h4 className={styles.titleProject}>{title}</h4>
            <p className={styles.subtitleProject}>{subtitle}</p>
            <Anchor url='/portfolio/[slug]' as={url} fontSize=".8rem" prefetch={false}>Detalles</Anchor>
        </div>
    )
}
export default ProjectListElement;