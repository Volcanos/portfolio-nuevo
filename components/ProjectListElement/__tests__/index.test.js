import React from 'react'
import { render } from 'enzyme';
import 'jest-styled-components'
import ProjectListElement from '../index';

export const content = {
    url: '/portfolio/push-your-kpis',
    mainImage: '/images/push1.png',
    title: 'Push Your KPIs',
    subtitle: 'Web',
    tags: ['Statamic', 'Laravel', 'React Native', 'Mobile', 'API', 'Notifications', 'Android', 'iOS'],
    web: {url: 'https://pushyourkpis.com', label: 'pushyourkpis.com'},
    description: 'El proyecto de Push Your KPIs consiste, en un administrador simple en que el usuario puede gestionar sus indicadores, y en una app para que los empleados puedan hacer seguimiento de esos mismos indicadores. Adem치s el gestor tiene la capacidad de enviar mensajes push filtrados para poder incentivar o informar a los empleados. Este proyecto se realiz칩 con el CMS Statamic que trabaja sobre Laravel. Para la aplicaci칩n se utiliz칩 React Native para el desarrollo completo de las aplicaciones para dispositivos Android e iOS.',
    listImages: [
        '/images/push2.png',
        '/images/push3.png',
    ]
}

describe('ProjectListElement', function () {
    it('should render title and subtitle', function () {
        const text = render(<ProjectListElement {...content} />).html();
        expect(text).toMatch(/Push Your KPIs/);
        expect(text).toMatch(/Web/);
    });
    it('should render image', function () {
        const text = render(<ProjectListElement {...content} />).html();
        expect(text).toMatch(/img/);
    });
    it('render snapshot not change', function () {
        const tree = render(<ProjectListElement {...content} />);
        expect(tree).toMatchSnapshot();
    });
});