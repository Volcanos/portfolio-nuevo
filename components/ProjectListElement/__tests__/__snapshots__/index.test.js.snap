// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListElement render snapshot not change 1`] = `
.c1 {
  display: block;
}

.c1:hover {
  cursor: pointer;
}

.c1:hover img {
  opacity: .8;
}

.c1 img {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c4 {
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c4:hover {
  color: #6f6f6f;
  cursor: pointer;
}

.c4:hover:before,
.c4:hover:after {
  width: 50%;
  opacity: 1;
}

.c4:before,
.c4:after {
  content: "";
  position: absolute;
  bottom: 0px;
  width: 0px;
  height: 1px;
  margin: 2px 0 0;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  -webkit-transition-duration: 0.65s;
  transition-duration: 0.65s;
  opacity: 0;
  background-color: #6f6f6f;
}

.c4:before {
  left: 50%;
}

.c4:after {
  right: 50%;
}

.c0 {
  margin-top: 2rem;
  margin-bottom: 5rem;
  text-align: center;
}

.c2 {
  color: #6f6f6f;
  font-weight: 800;
  margin-bottom: 0rem;
  margin-top: 1rem;
}

.c3 {
  color: #6f6f6f;
  font-size: 0.8rem;
  margin-bottom: 1rem;
}

<div
  class="c0"
>
  <a
    class="c1"
  >
    <img
      src="/images/push1.png"
      title="Push Your KPIs"
    />
  </a>
  <h4
    class="c2"
  >
    Push Your KPIs
  </h4>
  <p
    class="c3"
  >
    Web
  </p>
  <a
    class="c4"
  >
    Detalles
  </a>
</div>
`;
